Source/CarthageKit/Xcode.swift

```diff
@@ -1341,6 +1338,21 @@ public func copyProduct(from: NSURL, to: NSURL) -> SignalProducer<NSURL, Carthag
 	}
 }
 
+/// Copies existing files sent from the given producer into the given directory.
+///
+/// Returns a producer that will send locations where the copied files are.
+public func copyFileURLsFromProducer(producer: SignalProducer<NSURL, CarthageError>, intoDirectory directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
```

Comment:
I'm rather surprised you didn't make this a signal operator instead, e.g.

```swift
public func copyFileURLsIntoDirectory(directoryURL: NSURL) -> Signal<NSURL, CarthageError> -> Signal<NSURL, CarthageError>
```

because then you could have made the callers look more "functional":

```swift
let destinationDirectoryURL = frameworkURL.URLByDeletingLastPathComponent!
return dSYMForFramework(frameworkURL, inDirectoryURL:directoryURL)
        |> copyFileURLsIntoDirectory(destinationDirectoryURL)
```